.github/workflows/DiffTool-CI.yml

```diff
@@ -0,0 +1,140 @@
+          git config user.email "github-actions@github.com"
+          git config user.name "GitHub Actions"
+          git add .
+          git commit -m "Initialize diff-java-tool repository"
+
+      - name: Execute DiffTool.jar with config
+        env:
+          CONFIG_XML: "${{ github.workspace }}/${{ env.CONFIG_DIR }}/config.xml"
+          PROJECTS_YML: "${{ github.workspace }}/${{ env.CONFIG_DIR }}/project.yml"
+          DIFFTOOL_JAR_PATH: "${{ github.workspace }}/${{ env.DIFFTOOL_JAR }}"
+          PATCH_DIFF_TOOL_JAR_PATH: "${{ github.workspace }}/patch-diff-report-tool-${{ env.PATCH_DIFF_TOOL_VERSION }}-jar-with-dependencies.jar"
+        run: |
+          REPO="${{ github.workspace }}/checkstyle"
+          echo "Running DiffTool.jar in 'diff' mode"
+          java -jar "${DIFFTOOL_JAR_PATH}" \
+            --localGitRepo "$REPO" \
+            --baseBranch master \
+            --config "${CONFIG_XML}" \
+            --patchBranch temp-patch-branch \
+            --listOfProjects "${PROJECTS_YML}" \
+            --allowExcludes \
+            --diffToolJarPath "${PATCH_DIFF_TOOL_JAR_PATH}" || { echo "DiffTool execution failed"; exit 1; }
+
+      - name: Execute DiffTool.jar with patch config
+        env:
+          CONFIG_XML: "${{ github.workspace }}/${{ env.CONFIG_DIR }}/config.xml"
+          PATCH_CONFIG_XML: "${{ github.workspace }}/${{ env.CONFIG_DIR }}/patch-config.xml"
+          PROJECTS_YML: "${{ github.workspace }}/${{ env.CONFIG_DIR }}/project.yml"
+          DIFFTOOL_JAR_PATH: "${{ github.workspace }}/${{ env.DIFFTOOL_JAR }}"
+          PATCH_DIFF_TOOL_JAR_PATH: "${{ github.workspace }}/patch-diff-report-tool-${{ env.PATCH_DIFF_TOOL_VERSION }}-jar-with-dependencies.jar"
+        run: |
+          PR_BRANCH="temp-patch-branch"
+          REPO="${{ github.workspace }}/checkstyle"
+          echo "Running DiffTool.jar with all arguments combined"
+          java -jar "${DIFFTOOL_JAR_PATH}" \
+            --localGitRepo "$REPO" \
+            --baseBranch master \
+            --patchBranch "$PR_BRANCH" \
+            --baseConfig "${CONFIG_XML}" \
+            --patchConfig "${PATCH_CONFIG_XML}" \
+            --listOfProjects "${PROJECTS_YML}" \
+            --mode diff \
+            --shortFilePaths \
+            --extraMvnRegressionOptions "-Dmaven.test.skip=true" \
+            --allowExcludes \
+            --useShallowClone \
+            --diffToolJarPath "${PATCH_DIFF_TOOL_JAR_PATH}"
+
+      - name: Execute DiffTool.jar with Invalid Arguments
+        env:
+          DIFFTOOL_JAR_PATH: "${{ github.workspace }}/${{ env.DIFFTOOL_JAR }}"
+        run: |
+          echo "Running DiffTool.jar with invalid arguments"
+          java -jar "${DIFFTOOL_JAR_PATH}" --invalidArg
+          EXIT_CODE=$?
+          if [ "$EXIT_CODE" -eq 1 ]; then
+            echo "Error Exit code was expected."
+          else
+            echo "Unexpected exit code: $EXIT_CODE"
+            exit 1
+          fi
```

Comment:
\n please

Reply:
```
Run echo "Running DiffTool.jar with invalid arguments"
Running DiffTool.jar with invalid arguments
[main] INFO com.github.checkstyle.difftool.DiffTool - Unrecognized option: --invalidArg
usage: DiffTool
 -b,--baseBranch <arg>                   Base branch name. Default is
                                         master (optional, default is
                                         master)
 -bc,--baseConfig <arg>                  Path to the base checkstyle
                                         config file (optional, if absent
                                         then the tool will use only
                                         patchBranch in case the tool mode
                                         is 'single', otherwise baseBranch
                                         will be set to 'master')
 -c,--config <arg>                       Path to the checkstyle config
                                         file (required if baseConfig and
                                         patchConfig are not specified)
 -dt,--diffToolJarPath <arg>             Path to the
                                         patch-diff-report-tool JAR file
                                         (required)
 -g,--allowExcludes                      Whether to allow excludes
                                         specified in the list of projects
                                         (optional, default is false)
 -h,--useShallowClone                    Enable shallow cloning
 -l,--listOfProjects <arg>               Path to file which contains
                                         projects to test on (required)
 -m,--mode <arg>                         The mode of the tool: 'diff' or
                                         'single'. (optional, default is
                                         'diff')
 -p,--patchBranch <arg>                  Name of the patch branch in local
                                         git repository (required)
 -pc,--patchConfig <arg>                 Path to the patch checkstyle

```